src/annmas/utils/model.py

```diff
@@ -190,14 +191,57 @@ def build_default_model():
     return full_model
 
 
-def annotate(full_model, seq):
+def smooth(path, radius=5):
+    p = 0
+
+    while p < len(path):
+        # Are we in an adapter (as opposed to just the random model)?
+        if path[p] in adapters:
+            # We are!  How far does it extend?
+            p_adapter_left = p - 1
+            while p_adapter_left >= 0 and path[p_adapter_left] == path[p]:
+                p_adapter_left = p_adapter_left - 1
+
+            p_adapter_right = p + 1
+            while p_adapter_right < len(path) and path[p_adapter_right] == path[p]:
+                p_adapter_right = p_adapter_right + 1
+
+            # Now check if this is an adapter island (the surrounding states are just the random model)
+            prev_label = None
+            if p_adapter_left - radius >= 0:
+                prev_labels = set(path[(p_adapter_left - 1 - radius):(p_adapter_left - 1)])
+                if len(prev_labels) == 1:
+                    prev_label = next(iter(prev_labels))
```

Comment:
This is a little dangerous - sets don't have a guaranteed order.  I think `OrderedSet` would fix this.

Reply:
I'm not sure it matters - the goal is just to check if the previous (next) several states are the exact same value (hence the `len(prev_labels) == 1`, and that that value is 'random'.

Reply:
Ah.  Now I understand.  Yeah - this looks good.